### PR TITLE
[fuchsia] SceneHostBindings are no longer thread locals

### PR DESCRIPTION
**Pull request by iskakaushik**
Prior to this change SceneHostBindinds was a ThreadLocal but the
intention was for it to be IsolateLocal. Given that dart
could collect this map on a non-UI thread this caused
use-after-free issues.

This change fixes it by making it keyed on isolate and koid
this is not the ideal solution, this would exist on
dart isolate group data struct. Given that Fuchsia is moving
to use the embedder API, the decision to use this temporary
work around was made.

fixes https://github.com/flutter/flutter/issues/49738

### CODE REVIEW

File: lib/ui/compositing/scene_host.h [hunk 1]
```diff
@@ -12,12 +12,36 @@
...
 
 #include "dart-pkg/zircon/sdk_ext/handle.h"
+#include "dart/runtime/include/dart_api.h"
 #include "flutter/fml/memory/ref_counted.h"
 #include "flutter/fml/task_runner.h"
 #include "flutter/lib/ui/dart_wrapper.h"
 
 namespace flutter {
 
+struct SceneHostBindingKey {
```
Comment (arbreng) [inline lib/ui/compositing/scene_host.h]: Can you move all of this and the #include into the .cc file?

Comment (iskakaushik) [reply lib/ui/compositing/scene_host.h]: yup!

File: lib/ui/compositing/scene_host.cc [hunk 2]
```diff
@@ -10,33 +10,49 @@
...
+#include "dart/runtime/include/dart_api.h"
 #include "flutter/flow/view_holder.h"
 #include "flutter/fml/thread_local.h"
 #include "flutter/lib/ui/ui_dart_state.h"
 
 namespace {
 
-using SceneHostBindings = std::unordered_map<zx_koid_t, flutter::SceneHost*>;
+struct SceneHostBindingKey {
+  Dart_Isolate associated_isolate;
```
Comment (jason-simmons) [inline lib/ui/compositing/scene_host.cc]: Call `Dart_IsolateServiceId` and use that to represent the isolate.  A raw isolate handle can become invalid at any time.

Comment (iskakaushik) [reply lib/ui/compositing/scene_host.cc]: good call, thanks @jason-simmons

**Pull request merged**